server/src/test/java/org/opensearch/index/engine/NRTReplicationEngineTests.java

```diff
@@ -203,6 +203,36 @@ public void testSimultaneousEngineCloseAndCommit() throws IOException, Interrupt
         }
     }
 
+    public void testSimultaneousFlushAndCommit() throws IOException, InterruptedException {
+        final AtomicLong globalCheckpoint = new AtomicLong(SequenceNumbers.NO_OPS_PERFORMED);
+        try (
+            final Store nrtEngineStore = createStore(INDEX_SETTINGS, newDirectory());
+            final NRTReplicationEngine nrtEngine = buildNrtReplicaEngine(globalCheckpoint, nrtEngineStore)
+        ) {
+            CountDownLatch latch = new CountDownLatch(1);
+            Thread commitThread = new Thread(() -> {
+                try {
+                    nrtEngine.updateSegments(store.readLastCommittedSegmentsInfo());
+                    latch.countDown();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            });
+            Thread closeThread = new Thread(() -> {
+                try {
+                    latch.await();
+                    nrtEngine.flush(true, true);
```

Comment:
What does it mean to set `force` parameter to `true`? From the overridden flush function doesn't look like it's used in NRTEngine?

Reply:
The force param for writeable engines means to perform a commit even if there are no new changes.  With SR the flush will always be "forced" in this context because we do not write segments in this method, we only create a new commit point.